src/App.tsx

```diff
@@ -1,12 +1,74 @@
 import React from 'react';
 import './App.scss';
+import GoodsList from './GoodsList/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Product } from './types/Product';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  products: Product[] | null,
+  errorMessage: string,
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    products: null,
+    errorMessage: '',
+  };
+
+  getData = async (func: () => Promise<Good[]>) => {
+    try {
+      const goods = await func();
+
+      this.setState({ products: goods });
+    } catch (error) {
+      this.setState(
+        { errorMessage: 'Ooops! Something went wrong' },
+      );
+      throw new Error(`Error - ${error}`);
+    }
+  };
+
+  render() {
+    return (
+      <>
+        <h1>Dynamic list of Goods</h1>
+
+        <button
+          type="button"
+          onClick={() => {
+            this.getData(getAll);
+          }}
+        >
+          Load All goods
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            this.getData(get5First);
+          }}
+        >
+          Load 5 first goods
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            this.getData(getRedGoods);
+          }}
+        >
+          Load red goods
+        </button>
+
+        {this.state.products && (
+          <GoodsList goods={this.state.products} />
+        )}
+        <p>
+          {this.state.errorMessage}
+        </p>
```

Comment:
If there would be an error on the first try, user will see an error message even after successful loading

Reply:
Fix the last comment